File: COTE/LeetCode/medium/FindDiagonalOrder.py
# way to flat... 
# result=[[1],[2,3]] [e for arr in result for e in arr ]
class Solution:
    def findDiagonalOrder(self, mat: List[List[int]]) -> List[int]:
        m=len(mat)
        n=len(mat[0])
        result=[[] for i in range(m+n-1)]
        for i in range(0, m):
            for j in range(0, n):
                key=i+j
                if key%2==0:
                    result[key].insert(0,mat[i][j])
                else:
                    result[key].append(mat[i][j])
        return [e for arr in result for e in arr ]

File: COTE/LeetCode/medium/FindDiagonalOrder.js
// https://leetcode.com/explore/learn/card/array-and-string/202/introduction-to-2d-array/1167/
//✨✨✨
// careful with .fill -> push ->will have impact on all filled units
// new Array(m+n-1).fill(0).map(e=>[])
// change direction with key !
// arr.flat()
/**
 * @param {number[][]} mat
 * @return {number[]}
 */
 var findDiagonalOrder = function(mat) {
   let m=mat.length
   let n=mat[0].length
   let result = new Array(m+n-1).fill(0).map(e=>[])
   
   for(let i=0; i<m; i++){
       for(let j=0; j<n; j++){
           let key=i+j
           if(key%2===0){
               //upward
               result[key].unshift(mat[i][j])
           } else {
               //downward
               result[key].push(mat[i][j])
           }
       }
   }
   return result.flat();
};